From 61f87213ef6e3e59f08674a673229f97cd3ac40f Mon Sep 17 00:00:00 2001
From: vasrajan <vasrajan@cisco.com>
Date: Mon, 23 Sep 2019 13:43:48 -0700
Subject: [PATCH 3/4] Port from master - to prevent xcvrd crash due to missing
 fields

---
 sonic_sfp/sfputilbase.py | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/sonic_sfp/sfputilbase.py b/sonic_sfp/sfputilbase.py
index 3394fc2..a223fdc 100644
--- a/sonic_sfp/sfputilbase.py
+++ b/sonic_sfp/sfputilbase.py
@@ -760,7 +760,22 @@ class SfpUtilBase(object):
         transceiver_dom_info_dict = {}
 
         if port_num in self.osfp_ports:
-            pass
+            # Below part is added to avoid fail xcvrd, shall be implemented later
+            transceiver_dom_info_dict['temperature'] = 'N/A'
+            transceiver_dom_info_dict['voltage'] = 'N/A'
+            transceiver_dom_info_dict['rx1power'] = 'N/A'
+            transceiver_dom_info_dict['rx2power'] = 'N/A'
+            transceiver_dom_info_dict['rx3power'] = 'N/A'
+            transceiver_dom_info_dict['rx4power'] = 'N/A'
+            transceiver_dom_info_dict['tx1bias'] = 'N/A'
+            transceiver_dom_info_dict['tx2bias'] = 'N/A'
+            transceiver_dom_info_dict['tx3bias'] = 'N/A'
+            transceiver_dom_info_dict['tx4bias'] = 'N/A'
+            transceiver_dom_info_dict['tx1power'] = 'N/A'
+            transceiver_dom_info_dict['tx2power'] = 'N/A'
+            transceiver_dom_info_dict['tx3power'] = 'N/A'
+            transceiver_dom_info_dict['tx4power'] = 'N/A'
+
         elif port_num in self.qsfp_ports:
             offset = 0
             offset_xcvr = 128
-- 
2.7.4

