From b0ad8bcf0a33190c2e3e807c8b27d4d6aabb9c63 Mon Sep 17 00:00:00 2001
From: Jiahua Wang <jiahwang@cisco.com>
Date: Mon, 8 Oct 2018 15:10:54 -0700
Subject: [PATCH 4/4] change to patch/series file upto cisco-4.3

Signed-off-by: Jiahua Wang <jiahwang@cisco.com>
---
 patch/series | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/patch/series b/patch/series
index 3608593..c7d29a0 100644
--- a/patch/series
+++ b/patch/series
@@ -66,3 +66,13 @@ linux-4.13-thermal-intel_pch_thermal-Fix-enable-check-on.patch
 # Tdriver-hwmon-pmbus-ucd9200-mlnx.patch
 # Trtnetlink-catch-EOPNOTSUPP-errors.patch
 # Tbridge-per-port-multicast-broadcast-flood-flags.patch
+driver-mfd-enable-dh89xxcc-gpio.patch
+driver-rtc-cmos-address-space-256.patch
+driver-hwmon-adt7462-fix-alarm-mask.patch
+driver-hwmon-pmbus-cisco_n2200.patch
+driver-hwmon-pmbus-cisco_n2200-update.patch
+driver-hwmon-pmbus-cisco-solvan.patch
+include-asm-generic-gpio-ARCH_NR_GPIOs-1024.patch
+config-cisco-n3132-gx.patch
+config-cisco-n9200.patch
+driver-hwmon-pmbus-cisco-solvan-update.patch
-- 
2.7.4

