From 53bb82eef3daebd75c61265674224b51c495b718 Mon Sep 17 00:00:00 2001
From: aaliv <aaliv@cisco.com>
Date: Mon, 12 Jul 2021 02:27:26 -0700
Subject: [PATCH] Overrides-for-kconfig-inclusions-exclusion

Signed-off-by: aaliv <aaliv@cisco.com>
---
 patch/kconfig-inclusions |    4 ++++
 1 file changed, 4 insertions(+)

Index: sonic-linux-kernel/patch/kconfig-inclusions
===================================================================
--- sonic-linux-kernel.orig/patch/kconfig-inclusions
+++ sonic-linux-kernel/patch/kconfig-inclusions
@@ -56,6 +56,10 @@ CONFIG_MELLANOX_PLATFORM=y
 CONFIG_MLXREG_HOTPLUG=m
 CONFIG_MLXREG_IO=m
 CONFIG_MAX1363=m
+CONFIG_MTD=y
+CONFIG_MTD_SPI_NOR=y
+CONFIG_MTD_SPI_NOR_USE_4K_SECTORS=y
+CONFIG_SPI_INTEL_SPI_PCI=m
 
 # For Cisco 8000
 CONFIG_PHYLIB=m
