Index: sonic-py-common/sonic_py_common/device_info.py
===================================================================
--- sonic-py-common.orig/sonic_py_common/device_info.py
+++ sonic-py-common/sonic_py_common/device_info.py
@@ -170,10 +170,10 @@ def get_path_to_platform_dir():
     # Determine whether we're running in a container or on the host
     platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)
 
-    if os.path.isdir(CONTAINER_PLATFORM_PATH):
-        platform_path = CONTAINER_PLATFORM_PATH
-    elif os.path.isdir(platform_path_host):
+    if os.path.isdir(platform_path_host):
         platform_path = platform_path_host
+    elif os.path.isdir(CONTAINER_PLATFORM_PATH):
+        platform_path = CONTAINER_PLATFORM_PATH
     else:
         raise OSError("Failed to locate platform directory")
 
