diff --git a/tools/frr b/tools/frr
index 80dd9e8..369e7c7 100755
--- a/tools/frr
+++ b/tools/frr
@@ -87,11 +87,11 @@ check_daemon()
           if [ -n "$2" ]; then
            if [ ! -r "$C_PATH/$1-$2.conf" ]; then
                 touch "$C_PATH/$1-$2.conf"
-                chown frr:frr "$C_PATH/$1-$2.conf"
+                chown sonic:sonic "$C_PATH/$1-$2.conf"
            fi
           elif [ ! -r "$C_PATH/$1.conf" ]; then
             touch "$C_PATH/$1.conf"
-            chown frr:frr "$C_PATH/$1.conf"
+            chown sonic:sonic "$C_PATH/$1.conf"
           fi
         fi
         return 0
@@ -489,7 +489,7 @@ convert_daemon_prios
 if [ ! -d $V_PATH ]; then
     echo "Creating $V_PATH"
     mkdir -p $V_PATH
-    chown frr:frr $V_PATH
+    chown sonic:sonic $V_PATH
     chmod 755 /$V_PATH
 fi
 
