Index: sonic-utilities/ssdutil/main.py
===================================================================
--- sonic-utilities.orig/ssdutil/main.py
+++ sonic-utilities/ssdutil/main.py
@@ -32,19 +32,15 @@ def import_ssd_api(diskdev):
 
     # try to load platform specific module
     try:
-        platform_path, _ = device_info.get_paths_to_platform_and_hwsku_dirs()
-        platform_plugins_path = os.path.join(platform_path, "plugins")
-        sys.path.append(os.path.abspath(platform_plugins_path))
-        from ssd_util import SsdUtil
+        from sonic_platform_base.sonic_ssd.ssd_generic import SsdUtil
+        from ssd_util_wrapper import SsdInfo,ssd_get_params
+        ssd = SsdInfo()
+        ssd_get_params(ssd)
+        return SsdUtil(diskdev, ssd) 
     except ImportError as e:
         log.log_warning("Platform specific SsdUtil module not found. Falling down to the generic implementation")
-        try:
-            from sonic_platform_base.sonic_ssd.ssd_generic import SsdUtil
-        except ImportError as e:
-            log.log_error("Failed to import default SsdUtil. Error: {}".format(str(e)), True)
-            raise e
 
-    return SsdUtil(diskdev)
+    return SsdUtil(diskdev, None)
 
 def is_number(s):
     try:
