Index: sonic-utilities/scripts/reboot
===================================================================
--- sonic-utilities.orig/scripts/reboot
+++ sonic-utilities/scripts/reboot
@@ -9,7 +9,7 @@ REBOOT_TIME=$(date)
 VMCORE_FILE=/proc/vmcore
 if [ -e $VMCORE_FILE -a -s $VMCORE_FILE ]; then
         echo "We have a /proc/vmcore, then we just kdump'ed"
-        echo "User issued 'kdump' command [User: kdump, Time: ${REBOOT_TIME}]" > ${REBOOT_CAUSE_FILE}
+        echo "Kernel Panic [Time: ${REBOOT_TIME}]" > ${REBOOT_CAUSE_FILE}
         sync
         PLATFORM=$(grep -oP 'sonic_platform=\K\S+' /proc/cmdline)
         if [ ! -z "${PLATFORM}" -a -x ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} ]; then
@@ -50,6 +50,17 @@ function tag_images()
     fi
 }
 
+function stop_pmon_service()
+{
+     CONTAINER_STOP_RC=0
+     debug "Stopping pmon docker"
+     docker kill pmon &> /dev/null || CONTAINER_STOP_RC=$?
+     systemctl stop pmon || debug "Ignore stopping pmon error $?"
+     if [[ CONTAINER_STOP_RC -ne 0 ]]; then
+        debug "Failed killing container pmon RC $CONTAINER_STOP_RC ."
+     fi
+}
+
 function stop_sonic_services()
 {
     if [[ x"$ASIC_TYPE" != x"mellanox" ]]; then
@@ -57,6 +68,9 @@ function stop_sonic_services()
         docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
         sleep 3
     fi
+	
+	#Stop PMON docker before cold and soft reboots (pull #1514)
+	stop_pmon_service
 }
 
 function clear_warm_boot()
Index: sonic-utilities/show/reboot_cause.py
===================================================================
--- sonic-utilities.orig/show/reboot_cause.py
+++ sonic-utilities/show/reboot_cause.py
@@ -7,16 +7,21 @@ from tabulate import tabulate
 from swsssdk import SonicV2Connector
 import utilities_common.cli as clicommon
 
+PREVIOUS_REBOOT_CAUSE_FILE_PATH = "/host/reboot-cause/previous-reboot-cause.json"
 
-PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.json"
-USER_ISSUED_REBOOT_CAUSE_REGEX ="User issued \'{}\' command [User: {}, Time: {}]"
 
 def read_reboot_cause_file():
-    result = ""
-    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE):
-        with open(PREVIOUS_REBOOT_CAUSE_FILE) as f:
-            result = json.load(f)
-    return result
+    reboot_cause_dict = {}
+
+    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE_PATH):
+        with open(PREVIOUS_REBOOT_CAUSE_FILE_PATH) as prev_reboot_cause_file:
+            try:
+                reboot_cause_dict = json.load(prev_reboot_cause_file)
+            except json.JSONDecodeError as err:
+                click.echo("Failed to load JSON file '{}'!".format(PREVIOUS_REBOOT_CAUSE_FILE_PATH), err=True)
+
+    return reboot_cause_dict
+
 
 #
 # 'reboot-cause' group ("show reboot-cause")
@@ -26,15 +31,34 @@ def read_reboot_cause_file():
 def reboot_cause(ctx):
     """Show cause of most recent reboot"""
     if ctx.invoked_subcommand is None:
-        reboot_cause = ""
+        reboot_cause_str = ""
+
         # Read the previous reboot cause
-        data = read_reboot_cause_file()
-        if data['user'] == "N/A":
-            reboot_cause = "{}".format(data['cause'])
+        reboot_cause_dict = read_reboot_cause_file()
+
+        reboot_cause = reboot_cause_dict.get("cause", "Unknown")
+        reboot_user = reboot_cause_dict.get("user", "N/A")
+        reboot_time = reboot_cause_dict.get("time", "N/A")
+
+        if reboot_user != "N/A":
+            reboot_cause_str = "User issued '{}' command".format(reboot_cause)
         else:
-            reboot_cause = USER_ISSUED_REBOOT_CAUSE_REGEX.format(data['cause'], data['user'], data['time'])
+            reboot_cause_str = reboot_cause
+
+        if reboot_user != "N/A" or reboot_time != "N/A":
+            reboot_cause_str += " ["
+
+            if reboot_user != "N/A":
+                reboot_cause_str += "User: {}".format(reboot_user)
+                if reboot_time != "N/A":
+                    reboot_cause_str += ", "
+
+            if reboot_time != "N/A":
+                reboot_cause_str += "Time: {}".format(reboot_time)
+
+            reboot_cause_str += "]"
 
-        click.echo(reboot_cause)
+        click.echo(reboot_cause_str)
 
 # 'history' subcommand ("show reboot-cause history")
 @reboot_cause.command()
Index: sonic-utilities/tests/reboot_cause_test.py
===================================================================
--- sonic-utilities.orig/tests/reboot_cause_test.py
+++ sonic-utilities/tests/reboot_cause_test.py
@@ -29,7 +29,7 @@ class TestShowRebootCause(object):
 
     # Test 'show reboot-cause' without previous-reboot-cause.json 
     def test_reboot_cause_no_history_file(self):
-        expected_output = ""
+        expected_output = "Unknown\n"
         runner = CliRunner()
         result = runner.invoke(show.cli.commands["reboot-cause"], [])
         assert result.output == expected_output
