From ef760e9020a01ff06b2f59973d6f2f03e30f5d32 Mon Sep 17 00:00:00 2001
From: vasrajan <vasrajan@cisco.com>
Date: Tue, 12 May 2020 12:39:21 -0700
Subject: [PATCH] Temporary change for platform library load

---
 syncd/scripts/syncd_init_common.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/syncd/scripts/syncd_init_common.sh b/syncd/scripts/syncd_init_common.sh
index 169a8b1..71f65b5 100755
--- a/syncd/scripts/syncd_init_common.sh
+++ b/syncd/scripts/syncd_init_common.sh
@@ -162,6 +162,8 @@ config_syncd_innovium()
     export II_ROOT="/var/log/invm"
     export II_APPEND_LOG=1
     mkdir -p $II_ROOT
+
+    export PLATFORM_LIBRARY="/usr/lib/libinn_intf.so"
 }
 
 config_syncd()
-- 
2.7.4

